generate.js

```diff
@@ -42,7 +42,7 @@ function generate(args) {
     var targetOutputLocationList = []; // A list of objects that describe an sdk target
     ExtractArgs(args, argsByName, targetOutputLocationList, errorMessages);
 
-    if (!argsByName.buildidentifier) {
+    if (!argsByName.hasOwnProperty("buildidentifier")) {
```

Comment:
Usually I would prefer hasOwnProperty, as it's more explicit.
However, in this case, the former code evaluates as false for: (undefined, null, "")
The new code only checks undefined, and I think the former is a better test.